Packs/Base/Scripts/CommonServerPython/CommonServerPython.py

```diff
@@ -8028,33 +8016,34 @@ def total(self):
 
     @property
     def limit(self):
-        return self._next_limit
+        return self._limit
 
     @limit.setter
     def limit(self, value):
-        self._next_limit = self._original_limit = value
+        self._limit = value
 
-    def _is_search_done(self):
+    def is_search_done(self):
         """
-        Checks one of these conditions:
-        1. self.limit is set, and it's updated to be less or equal to zero
+        Return True if one of these conditions is met (else False):
+        1. self.limit is set, and it's updated to be less or equal to zero - return True
         2. for search_after if self.total was populated by a previous search, but no self._search_after_param
         3. for page if self.total was populated by a previous search, but page is too large
         """
-        if self._search_is_done:
-            return True
-
-        reached_limit = isinstance(self._next_limit, int) and self._next_limit <= 0
+        reached_limit = self.limit is not None and self.limit <= self._total_iocs_fetched
         if reached_limit:
+            demisto.debug("IndicatorsSearcher has reached its limit: {}".format(self.limit))
+            # update limit to match _total_iocs_fetched value
+            if self._total_iocs_fetched > self.limit:
+                self.limit = self._total_iocs_fetched
```

Comment:
@glicht Please notice this line.
We update the `self.limit` with `self._total_iocs_fetched` to handle cases where it's greater than the limit - this will allow updates to limit to affect further searches that might be done while updating limit (otherwise `limit` can be lower than `_total_iocs_fetched` even after increasing its value by the calling integration).